src/java.base/share/classes/java/security/Provider.java

```diff
@@ -1231,23 +1276,18 @@ private void parseLegacyPut(String name, String value) {
      */
     public Service getService(String type, String algorithm) {
         checkInitialized();
-
         // avoid allocating a new ServiceKey object if possible
         ServiceKey key = previousKey;
         if (key.matches(type, algorithm) == false) {
             key = new ServiceKey(type, algorithm, false);
             previousKey = key;
         }
-        Service s = null;
-        if (!serviceMap.isEmpty()) {
-            s = serviceMap.get(key);
-        }
+
+        Service s = serviceMap.get(key);
         if (s == null) {
-            synchronized (this) {
-                ensureLegacyParsed();
-                if (legacyMap != null && !legacyMap.isEmpty()) {
-                    s = legacyMap.get(key);
-                }
+            s = legacyMap.get(key);
+            if (s != null && !s.isValid()) {
+                legacyMap.remove(key, s);
```

Comment:
This hunk seems to be different from the hunk in the original changeset: https://github.com/openjdk/jdk/commit/9b747491de01fd011b09668a67113e80c2b7c708#diff-c2847a8d1997d3d28a13354b05d5e21ac6656e7e17f849bc198715099b9b5bd2L1204-R1286 -- and the follow-ups did not change it as well. Does this stack of patches contain some other patch?

Reply:
This block comes from the JFR patch I reverted and re-applied: [https://github.com/openjdk/jdk/commit/bc2af47e1e4e6e96020e03a60ce098ddd17f63ba#diff-c2847a8d1997d3d28a13354b05d5[…]f849bc198715099b9b5bd2R1286-R1290](https://github.com/openjdk/jdk/commit/bc2af47e1e4e6e96020e03a60ce098ddd17f63ba#diff-c2847a8d1997d3d28a13354b05d5e21ac6656e7e17f849bc198715099b9b5bd2R1286-R1290)

Reply:
All right, thanks.